packages/fx-core/src/component/generator/generatorAction.ts

```diff
@@ -177,39 +73,73 @@ export const unzipAction: GeneratorAction = {
   },
 };
 
-export const fetchTemplateFromLocalAction: GeneratorAction = {
-  name: GeneratorActionName.FetchTemplateZipFromLocal,
+export const LocalTemplateAction: GeneratorAction = {
+  name: GeneratorActionName.LocalTemplate,
   run: async (context: GeneratorContext) => {
     if (context.outputs?.length) {
       return;
     }
     context.logProvider.debug(`Fetching zip from local: ${JSON.stringify(context)}`);
-    context.fallback = true;
     const fallbackPath = path.join(getTemplatesFolder(), "fallback");
     const fileName = `${context.language!}.zip`;
     const zipPath: string = path.join(fallbackPath, fileName);
 
     const data: Buffer = await fs.readFile(zipPath);
-    context.zip = new AdmZip(data);
+    const zip = new AdmZip(data);
     context.outputs = await unzip(
-      context.zip,
+      zip,
       context.destination,
       context.fileNameReplaceFn,
       context.fileDataReplaceFn,
       context.filterFn
     );
+
     if (!context.outputs?.length) {
       throw new TemplateNotFoundError(context.name);
     }
   },
 };
 
-export const TemplateActionSeq: GeneratorAction[] = [
-  fetchTemplateZipFromSourceCodeAction,
-  fetchUrlForHotfixOnlyAction,
-  fetchZipFromUrlAction,
-  unzipAction,
-  fetchTemplateFromLocalAction,
-];
+async function determineTemplateSource(context: GeneratorContext) {
+  let url = "local";
+  if (!Boolean(templateConfig.useLocalTemplate)) {
+    const latestTag = await getTemplateLatestTag(
+      context.language!,
+      context.tryLimits,
+      context.timeoutInMs
+    );
+    const latestVer = latestTag.split("@")[1];
```

Comment:
It's better to use `latestTag.replace(tagPrefix, "")`.
You can update the function as
```
const latestVersion = getLatestTemplateVersion()
const localVersion = getLocalTemplateVersion()
if (semver.gt(latestVersion, localVersion) {
  getTemplateUrl(latestVersion, language);
}
```

Reply:
Yes, it will be clearer.